### PR TITLE
Optional params

### PR DESCRIPTION
**Pull request by bstillitano**

### CODE REVIEW

File: OrdermentumSDK/Routers/InvoiceRouter.swift [hunk 1]
```diff
@@ -57,8 +57,8 @@ public enum InvoiceRouter: URLRequestConvertible {
     //Parameters
     var parameters: [String: Any] {
         switch self {
-        case .getInvoices(let retailerId, let supplierId, let sortBy, let sortOrder, let pageNo):
-            return ["retailerId": retailerId, "supplierId": supplierId, "sortBy[\(sortBy)]": sortOrder, "pageNo": pageNo]
+        case .getInvoices(let dictionary):
```
Comment (johndagostino) [inline OrdermentumSDK/Routers/InvoiceRouter.swift:60]: you lose typing and the autocomplete using a dictionary

Comment (johndagostino) [reply OrdermentumSDK/Routers/InvoiceRouter.swift:60]: have a InvoiceRequestFilter object with known filters

Comment (johndagostino) [reply OrdermentumSDK/Routers/InvoiceRouter.swift:60]: that serializes to a Parameters object

Comment (bstillitano) [reply OrdermentumSDK/Routers/InvoiceRouter.swift:60]: @johndagostino this is non public, InvoiceService.swift provides the autocomplete

**Pull request merged**